java/vector/src/main/java/org/apache/arrow/vector/FieldVector.java

```diff
@@ -52,6 +52,13 @@
   void loadFieldBuffers(ArrowFieldNode fieldNode, List<ArrowBuf> ownBuffers);
 
   /**
+   * load data into vector using just the Buffers flatbuffer and backing ArrowBuf
+   * @param buffersIterator
+   * @param buf
+   */
+  void loadFieldBuffers(BuffersIterator buffersIterator, ArrowBuf buf);
```

Comment:
the point here is that I don't want to create ArrowFieldNode objects. I think it defeats the purpose of using FlatBuffer in the first place if we are just going to deserialize the flatbuffer message into objects and then pass the objects to the vectors to load. So I want to allow a method to load vectors without creating any more objects than necessary. I would be okay with either modifying the original method to not use ArrowFieldNode objects, or adding a new method. I am really ok with either approach. The new method that I introduced here doesn't make use of FieldNodes at all, but I want to rework it to use FieldNode, but with an iterator that wraps around the FlatBuffer message rather than pojos.